File: 019.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        queue<ListNode *> q; 
        int count  = 0; 
        ListNode * node = head; 
        
        while(count < n){
            count ++; 
            q.push(node); 
            node = node->next; 
        }
         
        ListNode * previous = nullptr; 
        while (node != nullptr){
            previous = q.front(); 
            q.pop() ;
            q.push(node); 
            node = node->next; 
        }
        
        q.pop() ; 
        if (previous == nullptr) return q.empty()? nullptr: q.front(); 
        previous->next = q.empty() ? nullptr: q.front();
        
        return head; 
        
    }
};

File: 019.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        numberOfElements = self.countElements(head) 
        toRemovePos = numberOfElements - n # lenght = 1, n  = 1, 0 
        startNode = ListNode(-1, head)
        self.removeElement(startNode, None, toRemovePos, -1)
        
        return startNode.next 
        
    
    def removeElement(self, node: Optional[ListNode], prevNode, n, currPos): 
        if node == None: 
            return  
        
        if n == currPos: 
            prevNode.next = node.next
            return 
        
        self.removeElement(node.next, node, n, currPos+1)
        
        
    def countElements(self, node: Optional[ListNode]): 
        if node == None: 
            return 0 
        
        
        return self.countElements(node.next) + 1
